knora/dsplib/utils/excel_to_json_lists.py

```diff
@@ -259,65 +289,66 @@ def validate_list_with_schema(json_list: str) -> bool:
 
     Returns:
         True if the list passed validation, False otherwise
-
     """
     current_dir = os.path.dirname(os.path.realpath(__file__))
     with open(os.path.join(current_dir, '../schemas/lists-only.json')) as schema:
         list_schema = json.load(schema)
 
     try:
-        validate(instance=json_list, schema=list_schema)
+        jsonschema.validate(instance=json_list, schema=list_schema)
     except jsonschema.exceptions.ValidationError as err:
         print(err)
         return False
     print('List passed schema validation.')
     return True
 
 
-def prepare_list_creation(excelfolder: str, listname: str, comments: dict):
+def prepare_list_creation(
+    excelfolder: str,
+    listname: Optional[str],
+    comments: dict[str, Any]
+) -> tuple[dict[str, Any], list[str]]:
     """
-    Gets the excelfolder parameter and checks the validity of the files. It then makes the root node for the list.
+    Create the list of Excel files that can be used to build a JSON list. Then, create the root node for the JSON list.
 
     Args:
         excelfolder: path to the folder containing the Excel file(s)
         listname: name of the list to be created
         comments: comments for the list to be created
 
     Returns:
-        rootnode (dict): The rootnode of the list as a dictionary
-        excel_files (list[str]): list of the Excel files to process
+        rootnode: The rootnode of the list as a dictionary
+        excel_files: list of the Excel files to process
     """
     # reset the global variables before list creation starts
-    global cell_names
-    global list_of_lists
+    global list_of_previous_node_names
+    global list_of_lists_of_previous_cell_values
 
-    list_of_lists = []
-    cell_names = []
+    list_of_previous_node_names = []
+    list_of_lists_of_previous_cell_values = []
 
     # check if the given folder parameter is actually a folder
     if not os.path.isdir(excelfolder):
-        print(excelfolder, 'is not a directory.')
+        print(excelfolder, ' is not a directory.')
         exit(1)
 
     # create a list with all excel files from the path provided by the user
-    excel_files = [filename for filename in glob.iglob(f'{excelfolder}/*.xlsx') if
-                   not os.path.basename(filename).startswith('~$')]
+    excel_files = [filename for filename in glob.iglob(f'{excelfolder}/*.xlsx')
+                   if not os.path.basename(filename).startswith('~$')
+                   and os.path.isfile(filename)]
 
-    # check if all excel_files are actually files
+    # log the files that can be used
```

Comment:
IMO, "log" would mean writing it to a log file (which is not the case).

Reply:
thanks for the hint, changed it back